src/Commands/CreateOrEditTaskParser.ts

```diff
@@ -30,12 +31,20 @@ export const taskFromLine = ({ line, path }: { line: string; path: string }): Ta
         return task;
     }
 
+    const { setCreatedDate } = getSettings();
+    let createdDate: moment.Moment | null = null;
+    if (setCreatedDate) {
+        console.dir(window.moment());
+        createdDate = window.moment();
+    }
```

Comment:
How does it know to only add the creation date when it is creating a task, and not when it is editing an existing task?

In testing, it does seem to work fine, but I can't see what in the code is causing that to happen, and am intrigued.

Reply:
Also, how does it manage to not overwrite any existing, earlier createdDate already in a task that is being edited?

Reply:
So logic is done via CreateOrEditTaskParser:
* if we already have task, then we parse in from line and return early - https://github.com/obsidian-tasks-group/obsidian-tasks/blob/e19af7f9f53a100ed63839f68a94739376c57d0e/src/Commands/CreateOrEditTaskParser.ts#L30
* If we went to newTask condition, than we forward createdDate to Task constructor - https://github.com/obsidian-tasks-group/obsidian-tasks/blob/e19af7f9f53a100ed63839f68a94739376c57d0e/src/Commands/CreateOrEditTaskParser.ts#L34